src/dialects/abstract/query-generator.js

```diff
@@ -2143,13 +2143,23 @@ export class AbstractQueryGenerator {
           && !isModelStatic(order[0].model)
           && !(typeof order[0] === 'string' && model && model.associations !== undefined && model.associations[order[0]])
         ) {
-          subQueryOrder.push(this.quote(order, model, '->', options));
+          // TESTS ONLY PASS IF THIS IS INCLUDED
+          this.quote(order, model, '->', options);
```

Comment:
It seems that quote is actually manipulating the first argument if it is an array. If it is, it automatically replaces the argument with the model field's `field` property. Can you replace this code with the following?

```
          // if the field is aliased, we want to push the alias instead of the real field
          const field = model.rawAttributes[order[0]]?.field || order[0];
          const subQueryAlias = this._getAliasForField(modelName, field, options);
```

I'm not entirely happy with this since we have now duplicated logic